### PR TITLE
Fix supported Python version

### PR DESCRIPTION
**Pull request by n1k61n**
Update Python Version in pyproject.toml
Remove Unused Dependencies
Add Missing Dependency (requests)
Migrate Dev Dependencies to the New Style
Ensure everything works by testing the environment
poetry run pytest

### CODE REVIEW

File: tests/test_conftest.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Copyright (C) 2024 by Siisurit e.U., Austria.
+# All rights reserved. Distributed under the MIT License.
+
+from dotenv import load_dotenv
+load_dotenv()
```
Comment (mcsken) [inline tests/test_conftest.py:5]: Can you explain why did you create a new test file that doesn't align with any module and that does nothing but load the environment variables?

This was not part of the requirements.

You mentioned in the PR "Ensure everything works by testing the environment", but nothing is being actually tested here. And no environment-variable-dependent test is needed to complete the requirements of this issue.

File: pyproject.toml [hunk 2]
```diff
@@ -37,18 +37,14 @@ packages = [
...
-rich = ">=9, <14"
-gitpython = "^3"
+python = ">=3.11, <3.14"
+
 pyyaml = "^6.0.2"
 python-dotenv = "^1.0.1"
-pytest-mock = "^3.14.0"
 pydantic = "^2.10.2"
 pyjwt = "^2.9.0"
 requests-mock = "^1.12.1"
```
Comment (mcsken) [inline pyproject.toml:46]: `requests-mock` should have been removed from the main dependencies when it was moved to dev dependencies.

File: pyproject.toml [hunk 3]
```diff
@@ -65,6 +61,11 @@ check_done = "check_done.command:main"
 "Changelog" = "https://github.com/siisurit/check_done/blob/main/CHANGELOG.md"
 "Issue Tracker" = "https://github.com/siisurit/check_done/issues"
 
+[tool.poetry.group.dev.dependencies]
+cryptography = "^44.0.0"
+requests-mock = "^1.12.1"
+python-dotenv = "^1.0.1"
+
```
Comment (mcsken) [inline pyproject.toml:68]: `coverage`, `pytest`, `pytest-cov`, `pre-commit` and `ruff` should have been migrated from the old-style dev dependencies group to the new one here, and the old dev dependencies section should have been as a result removed.

File: pyproject.toml [hunk 4]
```diff
@@ -19,7 +19,7 @@ repository = "https://github.com/siisurit/check_done.git"
 documentation = "https://github.com/siisurit/check_done/blob/main/README.md"
 keywords = ["check", "closed", "done", "done done", "finished", "issue", "task", "validate"]
 classifiers = [
-        "Development Status :: 4 - Beta",
+     "Development Status :: 4 - Beta",
```
Comment (mcsken) [inline pyproject.toml:22]: The original indentation was correct.

File: .idea/check_done.iml [hunk 5]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <module type="PYTHON_MODULE" version="4">
   <component name="NewModuleRootManager">
-    <content url="file://$MODULE_DIR$" />
-    <orderEntry type="jdk" jdkName="Poetry (check_done)" jdkType="Python SDK" />
+    <content url="file://$MODULE_DIR$">
+      <excludeFolder url="file://$MODULE_DIR$/.venv" />
+    </content>
+    <orderEntry type="jdk" jdkName="Python 3.12 (check_done)" jdkType="Python SDK" />
```
Comment (mcsken) [inline .idea/check_done.iml:7]: Contributors should either adhere to the project's IDE configuration by renaming the virtual environment used for the project to match the existing IDE project settings, or not commit this file. This should be reverted.

File: pyproject.toml [hunk 6]
```diff
@@ -37,18 +37,14 @@ packages = [
 ]
 
 [tool.poetry.dependencies]
-python = ">=3.10, <3.14"
-pygments = "^2"
-chardet = "^5"
-rich = ">=9, <14"
-gitpython = "^3"
+python = ">=3.11, <3.14"
+
```
Comment (mcsken) [inline pyproject.toml:41]: Unnecessary empty space.

**Comment by roskakori** [comment]
Rejecting in favor of the cleaner PR #41.

**Pull request closed without merging**